src/OpenLoco/Utility/String.cpp

```diff
@@ -12,6 +12,40 @@
 
 namespace OpenLoco::Utility
 {
+    // Case-insensitive logical string comparison function
+    int32_t strlogicalcmp(const char* s1, const char* s2)
```

Comment:
Change to use `std::string_view`?

Reply:
Hmm. Normally, I'd be for that, but the entire function body is working with C char pointers. I would rather not have to rewrite the function, considering it leans pretty heavily on C string library functions (`isdigit`, `toupper`, `strtoul`).

Reply:
You would have to use `from_chars` instead of strtoul but i think the other functions would work fine.

Reply:
> I would rather not have to rewrite the function

Why not?

Reply:
https://godbolt.org/z/YsKhhGPrT this is my attempt at recreating it with string_views

Reply:
Think you can even make it `constexpr` but no real need.

Reply:
@duncanspumpkin Thanks, please feel free to push it as a commit. :)

Reply:
done

Reply:
Note that this isn't utf8 friendly but I don't think that is really an issue atm but might be a consideration if we are porting this up to OpenRCT2.

Reply:
Urgh how annoying its not supported by the compiler versions we are using. We will need to revert my commit and stick to the original code.

Reply:
@duncanspumpkin here you go:

https://godbolt.org/z/8cqvWefKf

Reply:
okay updated it with that snippet